packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -287,12 +403,33 @@ export class CredentialsModule {
    *
    */
   public async acceptCredential(credentialRecordId: string) {
-    const credentialRecord = await this.credentialService.getById(credentialRecordId)
-    const connection = await this.connectionService.getById(credentialRecord.connectionId)
+    const record = await this.credentialService.getById(credentialRecordId)
+    const { message, credentialRecord } = await this.credentialService.createAck(record)
 
-    const { message } = await this.credentialService.createAck(credentialRecord)
-    const outboundMessage = createOutboundMessage(connection, message)
-    await this.messageSender.sendMessage(outboundMessage)
+    if (credentialRecord.connectionId) {
+      const connection = await this.connectionService.getById(credentialRecord.connectionId)
+      const outboundMessage = createOutboundMessage(connection, message)
+
+      await this.messageSender.sendMessage(outboundMessage)
+    }
+    // Use ~service decorator otherwise
+    else if (credentialRecord.credentialMessage?.service && credentialRecord.requestMessage?.service) {
```

Comment:
I'm not sure I fully understand. I've never heard that else if is an anti-pattern in JS.

However I'm that a big fan of this code. The if/else shouldn't be there in the first place. But until out-of-band I'd like to keep it there

Reply:
I mean this
```
    }
    // Use ~service decorator otherwise
    else if 
```
instead of 
```
    } else if {
      // Use ~service decorator otherwise
    }
```